StorageTests/TestSQLiteBookmarks.swift

```diff
@@ -77,6 +79,188 @@ class TestSQLiteBookmarks: XCTestCase {
+            "bookmark_other_folder", "http://example.org/2", "Bookmark in another folder (nok)", BookmarkRoots.ToolbarFolderGUID,
+            "bookmark_good_nonsynced", "http://example.org/3", "Bookmark 1 (ok)", BookmarkRoots.MobileFolderGUID,
+            "bookmark_good_synced", "http://example.org/4", "Bookmark 2 (nok)", BookmarkRoots.MobileFolderGUID,
+            "bookmark_in_buffer", "http://example.org/5", "Bookmark in buffer(nok)", BookmarkRoots.MobileFolderGUID,
+            "bookmark_good_additional", "http://example.org/6", "Bookmark additional 1", BookmarkRoots.MobileFolderGUID,
+            "bookmark_good_additional_over_limit", "http://example.org/7", "Bookmark additional 2", BookmarkRoots.MobileFolderGUID,
+        ]
+
+        let structureQuery =
+        "INSERT INTO \(TableBookmarksLocalStructure) (parent, child, idx) VALUES " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?), " +
+        "(?, ?, ?) "
+
+        let structureArgs: Args = [
+            BookmarkRoots.MobileFolderGUID, "folder123", 0,
+            "folder123", "bookmark123", 0,
+            BookmarkRoots.ToolbarFolderGUID, "bookmark_other_folder", 0,
+            BookmarkRoots.MobileFolderGUID, "bookmark_good_nonsynced", 1,
+            BookmarkRoots.MobileFolderGUID, "bookmark_good_synced", 2,
+            BookmarkRoots.MobileFolderGUID, "bookmark_in_buffer", 3,
+            BookmarkRoots.MobileFolderGUID, "bookmark_good_additional", 4,
+            BookmarkRoots.MobileFolderGUID, "bookmark_good_additional_over_limit", 5,
+        ]
+
+        let bufferQuery =
+        "INSERT INTO \(TableBookmarksBuffer) (guid, type, bmkUri, title, parentid, parentName, server_modified) VALUES " +
+        "(?, \(BookmarkNodeType.bookmark.rawValue), ?, ?, ?, '', ?) "
+
+        let bufferArgs: Args = [
+            "bookmark_in_buffer", "http://example.org/5", "Bookmark in buffer(nok)", BookmarkRoots.MobileFolderGUID, Date.now()
+        ]
+
+        let bufferStructureQuery =
+        "INSERT INTO \(TableBookmarksBufferStructure) (parent, child, idx) VALUES " +
+        "(?, ?, ?)"
+
+        let bufferStructureArgs: Args = [
+            "bookmark_in_buffer", "bookmark_in_buffer", 0, // It's its own parent because we are lazy.
+        ]
+
+        db.run([
+            (sql: localQuery, args: localArgs),
+            (sql: structureQuery, args: structureArgs),
+            (sql: bufferQuery, args: bufferArgs),
+            (sql: bufferStructureQuery, args: bufferStructureArgs),
+        ]).succeeded()
+
+        let expectation = self.expectation(description: #function)
+        bookmarks.getLocalBookmarksAdditions(limit: 2) >>== { results in
+            XCTAssertEqual(results.count, 2)
+            XCTAssertEqual(results.map { $0.guid }, ["bookmark_good_nonsynced", "bookmark_good_additional"])
+            expectation.fulfill()
```

Comment:
Just use `.value` to synchronously wait on the `Deferred` (which is what `succeeded()` does under the hood), then do your checks. Much simpler than using expectations.